cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -175,4 +175,124 @@ SensorBridge* MapBuilderBridge::sensor_bridge(const int trajectory_id) {
   return sensor_bridges_.at(trajectory_id).get();
 }
 
+visualization_msgs::MarkerArray MapBuilderBridge::GetTrajectoryNodesList() {
+  visualization_msgs::MarkerArray trajectory_nodes_list;
+
+  for (int trajectory_id = 0;
+       trajectory_id < map_builder_.num_trajectory_builders();
+       ++trajectory_id) {
+    const auto trajectory_nodes =
+        map_builder_.sparse_pose_graph()->GetTrajectoryNodes();
+
+    int i = 0;
+    for (const auto& node : trajectory_nodes) {
```

Comment:
Instead of range based loop and keeping `i` in addition to it: Use a loop index instead?